src/SuperSocket.Connection/PipeConnectionBase.cs

```diff
@@ -68,9 +70,17 @@ protected void UpdateLastActiveTime()
 
         public async override IAsyncEnumerable<TPackageInfo> RunAsync<TPackageInfo>(IPipelineFilter<TPackageInfo> pipelineFilter)
         {
-            var packagePipe = !Options.ReadAsDemand
-                ? new DefaultObjectPipe<TPackageInfo>()
-                : new DefaultObjectPipeWithSupplyControl<TPackageInfo>();
+            IObjectPipe<TPackageInfo> packagePipe;
+            if (Options.ReadAsDemand)
+            {
+                var defaultObjectPipe = new DefaultObjectPipeWithSupplyControl<TPackageInfo>();
+                _supplyController = defaultObjectPipe;
```

Comment:
It seems you don't have to do this. Because the packagePipe will be pass to KestrelPipeConnection eventually.